### PR TITLE
Fix endian.h compile error on OSX and use native byte-swap functions on BSD and OSX

### PR DESCRIPTION
**Pull request by snowyu**
first the endian.h include file postiion should be:

* machine/endian.h (osx)
* sys/endian.h (freebsd)

second there are no htobe32/htobe16 etc functions in endian.h

### CODE REVIEW

**Comment by ShadowNinja** [comment]
* You can't legally put things into the public domain in some jurisdictions.  Use something like CC0.
  * There Is a lot of extra whitespace here (eg, around `#if`s).
  * Names starting with double underscores are reserved for the compiler/library implementation.  Define unprefixed versions instead of the other way around.
  * In preprocessor lines, any indentation should be placed before the octothorpe.
  * `portable_endian.h` should be renamed to `endian.h` since headers are assumed to be portable unless explicitly specified otherwise.
  * Error messages should use proper english (start with a capital letter, end with punctuation, etc).

**Comment by est31** [comment]
About the licensing issue: The file seems to originate from [this gist](https://gist.github.com/panzi/6856583), perhaps we can ask @panzi to license it as MIT.

**Comment by panzi** [comment]
You can have those couple of lines of trivial code under whatever license you want. It is called `portable_endian.h` to avoid name conflicts.

**Comment by snowyu** [comment]
* `public domain` means you can do anything with such project/file.
* error message? it should not be added in the pr.

**Comment by nerzhul** [comment]
It's okay for me except the licence, maybe BSD 2 clause licence is better than public domain

**Comment by neoascetic** [comment]
Works for me on Maverics

**Comment by neoascetic** [comment]
@nerzhul if FreeBSD was also affected, why didn't we got a error from your CI earlier?

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -248,15 +248,25 @@ else()
...
-	find_library(XXF86VM_LIBRARY Xxf86vm)
-	mark_as_advanced(XXF86VM_LIBRARY)
-	set(CLIENT_PLATFORM_LIBS ${CLIENT_PLATFORM_LIBS} ${XXF86VM_LIBRARY})
 endif()
 
-check_include_files(endian.h HAVE_ENDIAN_H)
+if(APPLE)
+    set(ENDIAN_H "machine/endian.h")
+else()
+    if(FreeBSD)
```
Comment (est31) [inline src/CMakeLists.txt]: Perhaps an elseif is simpler here? It would spare a line.

**Comment by Zeno-** [comment]
@ShadowNinja  In preprocessor lines, any indentation should be placed before the octothorpe.

That is far from what is commonly accepted. The more general (which I agree with) "style" is that the # should be the first character on the line

**Comment by neoascetic** [comment]
Just compiled latest on Yosemite without any problem. @snowyu could you please provide steps you are doing to reproduce the error?

**Comment by ShadowNinja** [comment]
@Zeno-: However that's not how most of Minetest does it.  It makes it easier to adjust the alignment with editor tools.

@neoascetic [Re: BSD]: Before it simply failed to find `endian.h` and used the slower functions.

**Comment by neoascetic** [comment]
@ShadowNinja well, then we definitely need these super-fast native implementations for OSX! :)

**Comment by nerzhul** [comment]
using kernel functions is good, yes :).

Except some tabs it's okay for me

**Comment by Novatux** [comment]
Looks good (although some indents could be fixed).

**Comment by sapier** [comment]
"Public Domain" isn't a valid term in german jurisdiction result would be no license at all here in germany. As minetest itself is licensed lgpl this would be a major issue. I Added the WIP flag to show there's something to fix in it.

**Comment by panzi** [comment]
I'll sign any copyright assignment you want for that piece of trivial code that would probably be ruled by any court as not copyright-worthy (Schöpfungshöhe nicht erreicht).

**Comment by neoascetic** [comment]
@sapier it was WIP half of a year ago. It is not going to be done by @snowyu since he isn't commenting since April. That's why there are lot of PRs: discussions takes such big amount of time that authors of PRs loose their motivation to end the work. So it is better to do changes by yourself on merge or close the PR.

**Comment by paramat** [comment]
@snowyu could you update this?

**Comment by snowyu** [comment]
What for update? It's too long long ago.

**Comment by paramat** [comment]
Okay closing, feel free to open a new PR for this if you want.

**Comment by neoascetic** [comment]
I lol'd

**Pull request closed without merging**